src/types.ts

```diff
@@ -27,37 +27,69 @@ export interface Instance {
     account: string,
     bid: Bid
   ) => Promise<boolean>;
-  isZAuctionApprovedToTransferNft: (account: string) => Promise<boolean>;
-  getZAuctionSpendAllowance: (account: string) => Promise<ethers.BigNumber>;
+  isZAuctionApprovedToTransferNft: (
+    account: string,
+    contractAddress: string
+  ) => Promise<boolean>;
+  isZAuctionApprovedToTransferNftLegacy: (
+    account: string,
+    contractAddress: string
+  ) => Promise<boolean>;
   getZAuctionSpendAllowanceByBid: (
     account: string,
     bid: Bid
   ) => Promise<ethers.BigNumber>;
-  getTradeTokenAddress: () => Promise<string>;
+  getZAuctionSpendAllowance: (
+    paymentTokenAddress: string,
+    account: string
+  ) => Promise<ethers.BigNumber>;
+  getZAuctionSpendAllowanceLegacy: (
+    account: string
+  ) => Promise<ethers.BigNumber>;
+  getDefaultPaymentToken: () => Promise<string>;
+  setDefaultPaymentToken: (
+    defaultTokenAddress: string,
+    signer: ethers.Signer
+  ) => Promise<ethers.ContractTransaction>;
+  getDomainNetworkPaymentToken: (networkId: string) => Promise<string>;
+  setDomainNetworkPaymentToken: (
+    networkId: string,
+    domainNetworkToken: string,
+    signer: ethers.Signer
+  ) => Promise<ethers.ContractTransaction>;
+  getPaymentTokenForDomain: (domainTokenId: string) => Promise<string>;
   approveZAuctionSpendTradeTokensByBid: (
     bid: Bid,
     signer: ethers.Signer
   ) => Promise<ethers.ContractTransaction>;
   approveZAuctionSpendTradeTokens: (
+    paymentTokenAddress: string,
```

Comment:
It would easier if the user provides a `tokenId` instead of a payment token address. Or provide both functions.